pkg/exec/exec.go

```diff
@@ -100,20 +100,20 @@ func RunWithStdoutReader(cmd Cmd, readerFunc func(io.Reader) error) error {
 	if err != nil {
 		return err
 	}
-	defer pw.Close()
-	defer pr.Close()
 	cmd.SetStdout(pw)
+	defer pr.Close()
+	defer pw.Close()
```

Comment:
this change is bad, we should defer these immediately after checking the creation, if the call above panics for some reason and then the caller recovers, we won't close these.

Reply:
just messed it up trying to recover the original,